File: sdk-android/src/main/java/org/sinou/android/pydia/data/AccountRepository.java
package org.sinou.android.pydia.data;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class AccountRepository {

    private AccountDao accountDao;
    private LiveData<List<Account>> allAccounts;

    public AccountRepository(Application application) {
        AccountDatabase db = AccountDatabase.getDatabase(application);
        accountDao = db.accountDao();
        allAccounts = accountDao.listAccounts();
    }

    public LiveData<List<Account>> listAccounts() {
        return allAccounts;
    }


    public void insert(Account word) {
        new insertAsyncTask(accountDao).execute(word);
    }

    private static class insertAsyncTask extends AsyncTask<Account, Void, Void> {

        private AccountDao accountDao;

        insertAsyncTask(AccountDao dao) {
            accountDao = dao;
        }

        @Override
        protected Void doInBackground(final Account... params) {
            accountDao.insert(params[0]);
            return null;
        }
    }
}

File: sdk-android/src/main/java/org/sinou/android/pydia/data/AccountRepository.kt
package org.sinou.android.pydia.data

import androidx.annotation.WorkerThread
import androidx.lifecycle.LiveData

class AccountRepository(private val accountDao: AccountDao) {

    val accounts : List<Account> = accountDao.getAllAccounts()

    @WorkerThread
    suspend fun insert(account: Account) {
        accountDao.insert(account)
    }

    @WorkerThread
    suspend fun update(account: Account) {
        accountDao.update(account)
    }
}
